packages/ra-core/src/controller/ListController.tsx

```diff
@@ -302,19 +265,19 @@ export class UnconnectedListController extends Component<
         return query.filter || {};
     }
 
-    updateData(query?: any) {
-        const params = query || this.getQuery();
-        const { sort, order, page = 1, perPage, filter } = params;
+    updateData(props = this.props) {
+        const query = this.getQuery(props);
+
+        const { sort, order, page = 1, perPage, filter } = query;
         const pagination = {
-            page: parseInt(page, 10),
```

Comment:
This was here for a reason. Page and perPage can be strings. You said that TypeScript complained that the params are already numbers, but I don't see where that comes from.

Reply:
From the ListParams interface

Reply:
Done in the getQuery function